src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
Please take a look at the new code which is "more" asynchronous. While a) is possible, b) cannot be done using libcurl. There is a way to postpone the callback but this would postpone receipt of the headers as well whereas the HttpClient contract is that the task should be completed as soon as headers are received and a reader is not required for these. For a), the documentation recommends using curl_multi_socket_action over curl_multi_perform.